DigitalLearningSolutions.Web/Controllers/Register/RegisterAtNewCentreController.cs

```diff
@@ -266,31 +271,35 @@ public async Task<IActionResult> SummaryPost()
         }
 
         [HttpGet]
-        public IActionResult Confirmation()
+        public IActionResult Confirmation(
+            string candidateNumber,
+            bool approved,
+            bool userHasAdminAccountAtCentre,
+            int? centreId
+        )
         {
-            var candidateNumber = (string?)TempData.Peek("candidateNumber");
-            var approvedNullable = (bool?)TempData.Peek("approved");
-            var centreIdNullable = (int?)TempData.Peek("centreId");
-            var hasAdminAccountNullable = (bool?)TempData.Peek("userHasAdminAccountAtCentre");
-            TempData.Clear();
-
-            if (candidateNumber == null || approvedNullable == null || centreIdNullable == null ||
-                hasAdminAccountNullable == null)
+            if (centreId == null)
             {
                 return RedirectToAction("Index");
             }
 
-            var hasAdminAccount = (bool)hasAdminAccountNullable;
-            var approved = (bool)approvedNullable;
-            var centreId = (int)centreIdNullable;
+            var userId = User.GetUserIdKnownNotNull();
+
+            var (unverifiedPrimaryEmail, unverifiedCentreEmails) =
+                userService.GetUnverifiedEmailsForUser(userId);
+            var (_, centreName, unverifiedCentreEmail) = unverifiedCentreEmails.First(uce => uce.centreId == centreId);
 
-            var viewModel = new InternalConfirmationViewModel(
+            var model = new InternalConfirmationViewModel(
                 candidateNumber,
                 approved,
-                hasAdminAccount,
-                centreId
+                userHasAdminAccountAtCentre,
+                centreId,
+                unverifiedPrimaryEmail,

```

Comment:
This is an internal registration journey, so we only need to show the centre email if at all.

Reply:
Similar on RegisterInternalAdminController

Reply:
removed in both journeys